policy-controller/k8s/index/src/index.rs

```diff
@@ -1172,11 +1176,71 @@ impl PolicyIndex {
         authzs
     }
 
-    fn http_routes(&self, server_name: &str) -> HashMap<String, InboundHttpRoute> {
+    fn route_client_authzs(
+        &self,
+        route_name: &str,
+        authentications: &AuthenticationNsIndex,
+    ) -> HashMap<AuthorizationRef, ClientAuthorization> {
+        let mut authzs = HashMap::default();
+
+        for (name, spec) in self.authorization_policies.iter() {
+            // Skip the policy if it doesn't apply to the route.
+            if let authorization_policy::Target::HttpRoute(name) = &spec.target {
+                if name != route_name {
+                    tracing::trace!(
+                        ns = %self.namespace,
+                        authorizationpolicy = %name,
+                        route = %route_name,
+                        target = %name,
+                        "AuthorizationPolicy does not target HttpRoute",
+                    );
+                    continue;
+                }
+            } else {
+                continue;
+            }
+
+            tracing::trace!(
+                ns = %self.namespace,
+                authorizationpolicy = %name,
+                route = %route_name,
+                "AuthorizationPolicy targets HttpRoute",
+            );
+            tracing::trace!(authns = ?spec.authentications);
```

Comment:
nit, TIOLI: should this just be a single event?
```suggestion
            tracing::trace!(
                ns = %self.namespace,
                authorizationpolicy = %name,
                route = %route_name,
                authns = ?spec.authentications,
                "AuthorizationPolicy targets HttpRoute",
            );
```

Reply:
I'm not sure that this is an improvement. `authns` might contain several/many largish datastructures that will make the log line less readable.